src/containers/Layout/ScrollToTop/index.tsx

```diff
@@ -0,0 +1,35 @@
+import { ButtonWrapper, Icon } from 'components';
+import { useEffect, useState } from 'react';
+
+const ScrollToTop = () => {
+  const [isVisible, setIsVisible] = useState(false);
+  const viewportHeight = window.innerHeight;
+  const scrollElem = document.getElementById('page-container');
+
+  useEffect(() => {
+    const toggleVisibility = () => {
+      if (scrollElem?.scrollTop && scrollElem.scrollTop > viewportHeight) {
+        setIsVisible(true);
+      } else {
+        setIsVisible(false);
+      }
+    };
+
+    scrollElem?.addEventListener('scroll', toggleVisibility);
+
+    return () => scrollElem?.removeEventListener('scroll', toggleVisibility);
+  }, []);
```

Comment:
Because `scrollElem` is `null` on mount, you'll need to add it to the list of dependencies of the effect in order for it to work on scroll:
```suggestion
  }, [scrollElem]);
```